pvlib/shading.py

```diff
@@ -0,0 +1,193 @@
+"""
+The ``shading`` module contains functions that model module shading and the
+associated effects on PV module output
+"""
+
+import numpy as np
+import pandas as pd
+from pvlib.tools import sind, cosd
+
+
+def masking_angle(surface_tilt, gcr, height):
+    """
+    The elevation angle below which diffuse irradiance is blocked.
+
+    The ``height`` parameter determines how far up the module's surface to
+    evaluate the masking angle.  The lower the point, the steeper the masking
+    angle [1]_.  SAM uses a "worst-case" approach where the masking angle
+    is calculated for the bottom of the array (i.e. ``height=0``) [2]_.
+
+    Parameters
+    ----------
+    surface_tilt : numeric
+        Panel tilt from horizontal [degrees].
+
+    gcr : float
+        The ground coverage ratio of the array [unitless].
```

Comment:
I have a notion that `gcr` is sometimes defined as `gcr := width * cos(tilt) / pitch` for fixed-tilt arrays.  Is that a common usage?  If so, this parameter description and the one in `masking_angle_passias` should clarify that `width / pitch` is meant.

Reply:
In my experience the usual definition of ground coverage ratio is the ratio of module area to ground area, where the "ground area" is left somewhat ambiguous. That's the definition used in [Pvsyst](https://forum.pvsyst.com/viewtopic.php?t=2379), [Helioscope ](https://www.folsomlabs.com/content/resources/module-spacing.pdf) and `pvlib.tracking.SingleAxisTracker`. I think we should stick with `gcr = width / pitch`. I'd be interested to see where the ratio of projected horizontal width to pitch is used.